python/hail/expr/scan/__init__.py

```diff
@@ -0,0 +1,24 @@
+from ...expr import aggregators
+from functools import wraps, update_wrapper
+
+
+def scan_decorator(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        func = getattr(f, '__wrapped__')
+        af = func.__globals__['_agg_func']
+        setattr(af, 'as_scan', True)
+        res = f(*args, **kwargs)
+        setattr(af, 'as_scan', False)
+        return res
+    update_wrapper(wrapper, f)
+    return wrapper
+
+
+__all__ = [name for name in dir(aggregators) if name[0] != '_']
+
+
+for name in __all__:
+    globals()[name] = scan_decorator(getattr(aggregators, name))
```

Comment:
this isn't something we should be doing. Python provides no guarantee that mutating the globals() dict actually affects the current frame, much less any other.

Reply:
really what we want is this: https://www.python.org/dev/peps/pep-0562/ but people might be unhappy if we pull the 3.6 rug out from under them :)

Reply:
I'm not sure what the best solution is. I'd be happy just listing out all the aggregators:
```python
count = decorate(hl.agg.count)
sum = decorate(hl.agg.sum)
```

Reply:
how about just an explicit setattr() on the scan module object? I could list out all the aggregators, but then maybe it would be better to just define them explicitly?